modules/adspiritBidAdapter.js

```diff
@@ -0,0 +1,72 @@
+import * as utils from 'src/utils';
+import {registerBidder} from 'src/adapters/bidderFactory';
+const BIDDER_CODE = 'adspirit';
+const RTB_URL = '/rtb/getbid.php?rtbprovider=prebid';
+const SCRIPT_URL = '/adasync.min.js';
+export const spec = {
+  code: BIDDER_CODE,
+  aliases: ['xapadsmedia'],
+  isBidRequestValid: function(bid) {
+    var host = spec.getBidderHost(bid);
+    if (!host) return false;
+    if (!bid.params.placementId) return false;
+    return true;
+  },
+  buildRequests: function(validBidRequests) {
+    var requests = [];
+    var bidRequest;
+    var reqUrl;
+    var placementId;
+    for (var i = 0; i < validBidRequests.length; i++) {
+      bidRequest = validBidRequests[i];
+      bidRequest.adspiritConId = spec.genAdConId(bidRequest);
+      reqUrl = spec.getBidderHost(bidRequest);
+      placementId = utils.getBidIdParameter('placementId', bidRequest.params);
+      reqUrl = '//' + reqUrl + RTB_URL + '&pid=' + placementId + '&ref=' + encodeURIComponent(utils.getTopWindowUrl()) + '&scx=' + (screen.width) + '&scy=' + (screen.height) + '&wcx=' + ('innerWidth' in window ? window.innerWidth : page.clientWidth) + '&wcy=' + ('innerHeight' in window ? window.innerHeight : page.clientHeight) + '&async=' + bidRequest.adspiritConId + '&t=' + Math.round(Math.random() * 100000);
+      requests.push({
+        method: 'GET',
+        url: reqUrl,
+        data: {},
+        bidRequest: bidRequest
+      });
+    }
+    return requests;
+  },
+  interpretResponse: function(serverResponse, bidRequest) {
+    const bidResponses = [];
+    var bidObj = bidRequest.bidRequest;
+
+    if (!serverResponse || serverResponse.status !== 'SUCCESS' || !bidObj) {
+      utils.logWarn(`No valid bids from ${spec.code} bidder`);
+      return [];
+    }
+    var cpm = serverResponse.cpm;
+    if (!cpm) return [];
+
+    var host = spec.getBidderHost(bidObj);
+    var adm = '<scr' + 'ipt>window.inDapIF=false</scr' + 'ipt><scr' + 'ipt src="//' + host + SCRIPT_URL + '"></scr' + 'ipt>' + '<ins id="' + bidObj.adspiritConId + '"></ins>' + serverResponse.adm;
+    const bidResponse = {
+      requestId: bidRequest.bidId,
+      bidderCode: spec.code,
+      cpm: cpm,
+      width: serverResponse.w,
+      height: serverResponse.h,
+      creativeId: serverResponse.placement_id,
+      currency: 'EUR',
+      netRevenue: true,
+      ad: adm
+    };
```

Comment:
This object should also set a `ttl`, see table in http://prebid.org/dev-docs/bidder-adapter-1.html#interpreting-the-response for description of this field